plugins/com.google.cloud.tools.eclipse.appengine.compat/src/com/google/cloud/tools/eclipse/appengine/compat/GpeMigrator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.compat;
+
+import com.google.cloud.tools.eclipse.util.NatureUtils;
+import java.util.Set;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+import org.eclipse.wst.common.project.facet.core.IProjectFacet;
+import org.eclipse.wst.common.project.facet.core.IProjectFacetVersion;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
+import org.eclipse.wst.common.project.facet.core.runtime.IRuntime;
+
+public class GpeMigrator {
+
+  private static final String GPE_GAE_NATURE_ID = "com.google.appengine.eclipse.core.gaeNature";
+
+  private static final String GPE_WTP_GAE_FACET_ID = "com.google.appengine.facet";
+  private static final String GPE_WTP_GAE_FACET_VERSION = "1";
+  private static final String GPE_WTP_GAE_EAR_FACET_ID = "com.google.appengine.facet.ear";
+  private static final String GPE_WTP_GAE_EAR_FACET_VERSION = "1";
+
+  private static final String GPE_WTP_GAE_RUNTIME = "com.google.appengine.runtime.id";
+
+  public static void removeObsoleteGpeFixtures(
```

Comment:
> fixtures? Is that the right word?
Not really. Just something that came up in my mind. Any suggestion that can accommodate facets, natures, and runtimes?